### PR TITLE
En garde! (nah just sword pointing)

### PR DESCRIPTION
**Pull request by silicons**
## About The Pull Request

You can now point the tip of some more swords/sharp tipped things (including spears, for the revolutionaries out there) at people.

## Why It's Good For The Game

En garde!

## Changelog
:cl:
tweak: You can now right click to point the tip of some sharp tipped weapons at people.
/:cl:

### CODE REVIEW

**Comment by Arturlang** [comment]
Wouldnt it make more sense if you were holding an item and if you pointed while holding and it being selected just pointing with the item always, when normally pointing?

**Comment by silicons** [comment]
@Arturlang yes but this is just an advanced webeditor PR 
can easily be switched to that since it's an element (and some things wouldn't make sense why would you for example point with a toolbox lmao)

File: code/game/objects/items/twohanded.dm [hunk 1]
```diff
@@ -605,6 +608,8 @@
 /obj/item/twohanded/spear/Initialize()
 	. = ..()
 	AddComponent(/datum/component/butchering, 100, 70) //decent in a pinch, but pretty bad.
+	AddComponent(/datum/component/josuting)
```
Comment (Kraseo) [inline code/game/objects/items/twohanded.dm]: josuting

Comment (lolman360) [reply code/game/objects/items/twohanded.dm]: kveinz000

File: code/game/objects/items/melee/energy.dm [hunk 2]
```diff
@@ -105,6 +104,10 @@
 	armour_penetration = 35
 	block_chance = 50
 
+/obj/item/melee/transforming/energy/sword/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/sword_point)
```
Comment (Ghommie) [inline code/game/objects/items/melee/energy.dm]: should only be doneable when active. Also remove lines 273-276.

File: code/datums/elements/sword_point.dm [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+		return ELEMENT_INCOMPATIBLE
+	RegisterSignal(target, COMSIG_ITEM_ALT_AFTERATTACK, .proc/point)
+
+/datum/element/sword_point/Detach(datum/source)
+	. = ..()
+	UnregisterSignal(source, COMSIG_ITEM_ALT_AFTERATTACK)
+
+/datum/element/sword_point/proc/point(datum/source, atom/target, mob/user, proximity_flag, params)
+	if(!proximity_flag && ismob(target))
+		user.visible_message("<span class='warning'>[user] points the tip of [source] at [target]!</span>")
```
Comment (Ghommie) [inline code/datums/elements/sword_point.dm]:
```suggestion
		user.visible_message("<span class='notice'>[user] points the tip of [src] at [target].</span>", "<span class='notice'>You point the tip of [src] at [target].</span>")
```

File: code/game/objects/items/twohanded.dm [hunk 4]
```diff
@@ -492,6 +494,7 @@
 /obj/item/twohanded/dualsaber/hypereutactic/ComponentInitialize()
 	. = ..()
 	AddElement(/datum/element/update_icon_updates_onmob)
+	AddElement(/datum/element/sword_point)
```
Comment (Ghommie) [inline code/game/objects/items/twohanded.dm]: Remove, this is a subtype of `/obj/item/twohanded/dualsaber` anyway.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]

**Pull request merged**